habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -115,38 +112,44 @@ def __init__(
         ngroups: int = 32,
         spatial_size: int = 128,
         make_backbone=None,
-        normalize_visual_inputs: bool = False,
     ):
         super().__init__()
 
         # Determine which visual observations are present
-        self.rgb_keys = [k for k in observation_space.spaces if "rgb" in k]
-        self.depth_keys = [k for k in observation_space.spaces if "depth" in k]
-
-        # Count total # of channels for rgb and for depth
-        self._n_input_rgb, self._n_input_depth = [
-            # sum() returns 0 for an empty list
-            sum(observation_space.spaces[k].shape[2] for k in keys)
-            for keys in [self.rgb_keys, self.depth_keys]
+        self.visual_keys = [
+            k
+            for k, v in observation_space.spaces.items()
+            if len(v.shape) > 1
+            and k != ImageGoalSensor.cls_uuid
+            and "debug" not in k
```

Comment:
No, the key is `robot_third_rgb` if I read the code correctly. I was thinking that we might eventually create more sensors that are used only for debugging (like we did in gala_kinematic) and that it would be nice to catch them here.
This is not used right now, should we remove it?

Reply:
I think this is the right thing to do (have some key that ignores sensors). However, how is the robot_third_rgb key being ignored by the policy right now?

Reply:
If my understanding is correct : Keys in `VIDEO_RENDER_VIEWS` are added at evaluation time only and *AFTER* the ResNetEncoder has been initialized. This means that `ResNetEncoder.visual_keys` is already fixed and adding new keys in the `ResNetEncoder.forward` arguments will not change the way it iterates over the observations.

Reply:
This still kinda hacky (having debug in the name), wouldn't it be better to add some attribute to the base sensor object in habitat-lab? Also, should this really be a list or an ordered set (do we allow duplicates)?

Reply:
We can remove the debug string since we already have VIDEO_RENDER_VIEWS. I agree it is a bit hacky. I don't think the ResNetEncoder init method should have access to the sensors, it currently only has the observation space and it should be enough. Bear in mind that this change is meant to allow non habitat environments to be trained, so we cannot rely on properties of habitat sensors. Given how the visual_keys list is constructed, there cannot be duplicates since the keys are generated from the keys of a dictionary. We can maje it an ordered set to stress that fact, but I don't think it is needed.